assimilation_code/programs/obs_sequence_tool/obs_sequence_tool.f90

```diff
@@ -8,21 +8,30 @@
 
 program obs_sequence_tool
 
-use        types_mod, only : r8, missing_r8, metadatalength, obstypelength
+use        types_mod, only : r8, missing_r8, metadatalength, obstypelength, &
+                             MISSING_I
+
 use    utilities_mod, only : finalize_utilities, initialize_utilities, &
                              find_namelist_in_file, check_namelist_read, &
                              error_handler, E_ERR, E_MSG, nmlfileunit,   &
-                             do_nml_file, do_nml_term, set_filename_list
+                             do_nml_file, do_nml_term, set_filename_list, &
+                             to_upper
+
 use     location_mod, only : location_type, get_location, set_location, &
                              LocationName !%! , vert_is_height 
-                             ! see comment in select_gps_by_height() for explanation of !%!
+                             ! comment in select_gps_by_height() explains !%!
+
 use      obs_def_mod, only : obs_def_type, get_obs_def_time, get_obs_def_type_of_obs, &
                              get_obs_def_location, set_obs_def_write_external_FO
-use     obs_kind_mod, only : max_defined_types_of_obs, get_name_for_type_of_obs, get_index_for_type_of_obs
+
+use     obs_kind_mod, only : max_defined_types_of_obs, get_name_for_type_of_obs, &
+                             get_index_for_type_of_obs
+
 use time_manager_mod, only : time_type, operator(>), print_time, set_time, &
                              print_date, set_calendar_type, GREGORIAN
+
 use obs_sequence_mod, only : obs_sequence_type, obs_type, write_obs_seq, &
-                             init_obs, assignment(=), get_obs_def, &
+                             init_obs, assignment(=), get_obs_def, set_obs_def, &
```

Comment:
After you've used the `set_obs_def_write_external_FO` routine, you need to be able to set the observation value to the new obs_def, so you need 'set_obs_def'